src/main/java/com/opentable/service/ServiceInfo.java

```diff
@@ -14,11 +14,151 @@
 
 package com.opentable.service;
 
+import java.net.URI;
+import java.util.Objects;
+
 /**
  * The {@link ServiceInfo} interface is intended for you to declare as a Spring Bean in your application.
  * It can be consumed by our components in order to create things like metric namespaces given information about your
  * service.  You are expected to implement your own implementations of this interface when defining your Bean.
+ * The easiest way to do so is use the Builder.
+ *
+ * For more details on these fields see https://wiki.otcorp.opentable.com/display/CP/The+service-details+Endpoint
```

Comment:
Additionally, making this part of code requires a parallel construction for each supported environment -- whereas if it is simply defined metadata, it is language independent.

Finally, how shall a service that does not serve HTTP (say, a memcached) give its service info?

Reply:
1. That's a valid and interesting point - maybe one could have this all defined in the config properties?
2. I don't think that's actually in score for ServiceDetails @strykerd

Reply:
Yeah, but if we define it in a matrix in the configuration file... we should just centralize the configuration in a file somewhere :/

Reply:
Going with #1 wouldn't eliminate this PR. Instead this PR would get merged and in otj-spring or otj-server (not sure where is appropriate, probably otj-serve) a springified configuration that loads from properties would be used. It would fail for missing service name or description, everything else it would just log a warning. And THAT could be composed into @RestServer

Reply:
If we don't expect something like `memcached` to expose service details, who exactly do we expect to expose service details?  It can't be everything with a `component-id` (since that could include a `memcached` or a headless queue worker) and definitions such as "everything that speaks HTTP and happens to have this endpoint" doesn't seem particularly useful...

Reply:
Valid questions that we should discuss with David that this PR is meant to spark ;)